python/tests/test_pyvw.py

```diff
@@ -192,4 +193,95 @@ def test_parse():
 
     ex = model.parse(["| a:1 b:0.5", "0:0.1:0.75 | a:0.5 b:1 c:2"])
     assert len(ex) == 2
+
+    # To test the prediction on multiline examples
+    finish = model.finish_example(ex)
+    assert model.predict(ex) == [0.0, 0.0]
     del model
+
+def test_namespace_id():
+    vw_ex = vw(quiet=True)
+    ex = vw_ex.example('1 |a two features |b more features here')
+    nm1 = pyvw.namespace_id(ex, 2)
+    assert nm1.id == 2
+    assert nm1.ord_ns == 128
+    assert nm1.ns == '\x80' # Represents string of ord_ns
+
+def test_example_namespace():
+    vw_ex = vw(quiet=True)
+    ex = vw_ex.example('1 |a two features |b more features here')
+    nmid = pyvw.namespace_id(ex, 1)
+    exm = pyvw.example_namespace(ex, nmid, ns_hash=vw_ex.hash_space(nmid.ns))
```

Comment:
The naming of these objects is unclear, please give them more descriptive names

Reply:
Okay, I will make it more clearer.